bookman

```diff
@@ -89,16 +89,16 @@ if [ -n "$SOURCE_DATE_EPOCH" ]; then
 fi
 date=${date:-$(date +'%d %B %Y')}
 
-[[ $1 ]] || set -- $(while read; do echo $REPLY; done)
+[ $1 ] || set -- $(while read REPLY; do echo $REPLY; done)
 
-[[ $outfile ]] && post="$post >$outfile"
+[ $outfile ] && post="$post >$outfile"
```

Comment:
Hi Marc,

Thanks for your quick reply.

You are right in all considerations. I thought a bit strange supress the
double quotes but I followed the bug. Really, use variables inside double
quotes is a good practice (I do it in my scripts) and not a bashism (I
searched for this now). Maybe an old 'checkbashisms' script in Debian has
caused this mistake.

The current checkbashisms says:

$ checkbashisms bookman
possible bashism in bookman line 92 (read without variable):
[[ $1 ]] || set -- $(while read; do echo $REPLY; done)
possible bashism in bookman line 92 (alternative test command ([[ foo ]]
should be [ foo ])):
[[ $1 ]] || set -- $(while read; do echo $REPLY; done)
possible bashism in bookman line 94 (alternative test command ([[ foo ]]
should be [ foo ])):
[[ $outfile ]] && post="$post >$outfile"

I will send a new pool request now. I will also fix the
improve-manpage-fix-spelling pull request.

Thanks! Thanks!

Regards,

Eriberto


2016-05-17 9:53 GMT-03:00 Marc Vertes <notifications@github.com>:

> In bookman
> <https://github.com/mvertes/txt2man/pull/5#discussion_r63515658>:
>
> >
> > -[[ $outfile ]] && post="$post >$outfile"
> > +[ $outfile ] && post="$post >$outfile"
>
> Same here: should be [ "$outfile" ] && post="$post >\"$outfile\"" in
> order to handle spaces in outfile value
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mvertes/txt2man/pull/5/files/af27afaa0cc6a8664a9c6b0f17ce42f9e90d12fe#r63515658>
>